symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java

```diff
@@ -19,120 +24,200 @@
+    assertEquals(config.getBot().getPrivateKeyPath(), "/Users/local/conf/agent/privatekey.pem");
+    assertEquals(config.getApp().getPrivateKeyPath(), "/Users/local/conf/agent/privatekey.pem");
+  }
+
+  @Test
+  void loadFromClasspathNotFoundTest() {
+    BdkConfigException exception = assertThrows(BdkConfigException.class, () -> {
+      BdkConfigLoader.loadFromClasspath("/wrong_classpath/config.yaml");
+    });
+    assertEquals(exception.getMessage(), "Config file is not found");
+  }
+
+  @Test
+  void loadClientGlobalConfig() throws BdkConfigException {
+    BdkConfig config = BdkConfigLoader.loadFromClasspath("/config/config_client_global.yaml");
+
+    assertEquals(config.getPod().getScheme(), "https");
+    assertEquals(config.getPod().getHost(), "diff-pod.symphony.com");
+    assertEquals(config.getPod().getPort(), 8443);
+    assertEquals(config.getPod().getContext(), "context");
+
+    assertEquals(config.getScheme(), "https");
+
+    assertEquals(config.getAgent().getScheme(), "https");
+    assertEquals(config.getAgent().getHost(), "devx1.symphony.com");
+    assertEquals(config.getAgent().getPort(), 443);
+    assertEquals(config.getAgent().getFormattedContext(), "/context");
+
+    assertEquals(config.getKeyManager().getScheme(), "https");
+    assertEquals(config.getKeyManager().getHost(), "devx1.symphony.com");
+    assertEquals(config.getKeyManager().getPort(), 8443);
+    assertEquals(config.getKeyManager().getFormattedContext(), "/diff-context");
+
+    assertEquals(config.getSessionAuth().getScheme(), "http");
+    assertEquals(config.getSessionAuth().getHost(), "devx1.symphony.com");
+    assertEquals(config.getSessionAuth().getPort(), 8443);
+    assertEquals(config.getSessionAuth().getContext(), "context");
+  }
+
+
+  @Test
+  public void parseLbAgentField() throws BdkConfigException {
+    BdkConfig config = BdkConfigLoader.loadFromClasspath("/config/config_lb.yaml");
+    final BdkLoadBalancingConfig agentLoadBalancing = config.getAgentLoadBalancing();
+    final List<BdkServerConfig> nodes = agentLoadBalancing.getNodes();
+
+    assertEquals(BdkLoadBalancingMode.RANDOM, agentLoadBalancing.getMode());
+    assertEquals(false, agentLoadBalancing.isStickiness());
+    assertEquals(2, nodes.size());
+
+    assertEquals("http", nodes.get(0).getScheme());
+    assertEquals("agent1.acme.org", nodes.get(0).getHost());
+    assertEquals(8443, nodes.get(0).getPort());
+    assertEquals("/app", nodes.get(0).getContext());
+
+    assertEquals("https", nodes.get(1).getScheme());
+    assertEquals("agent2.acme.org", nodes.get(1).getHost());
+    assertEquals(443, nodes.get(1).getPort());
+    assertEquals("", nodes.get(1).getContext());
+  }
+
+  @Test
+  public void parseLbAgentFieldsWithNoDefinedStickiness() throws BdkConfigException {
+    BdkConfig config = BdkConfigLoader.loadFromClasspath("/config/config_lb_no_stickiness.yaml");
+    assertEquals(true, config.getAgentLoadBalancing().isStickiness());
```

Comment:
assertTrue /assertFalse can be used directly

also public modified can be remove with junit 5 (I recommend installing sonarlint plugin in idea for that)